### DISCUSSION TITLE
xbpeng release adversarial motion priors : Which Bullet Physics release to use?

### DISCUSSION THREAD

**Discussion by mellogrand** [bulletphysics/bullet3#3862] [General]
Sorry for possibly distirbing you,  but in your bulletphysics/bullet3/releases in the **last 3.17 version you say: This updated Bullet Physics release was used for the "Learning Agile Robotic Locomotion Skills by Imitating Animals" research., led by Jason Peng.** 

**In the last version of Jason's DeepMimic on Github** , updated on 6 Apr 2021 for its last paper "AMP: Adversarial Motion Priors for Stylized Physics-Based Character Control" 
and referring also to the previous "DeepMimic: Example-Guided Deep Reinforcement Learning of Physics-Based Character Skills" instructions  still say
**Download Bullet 2.88 from the above link and install using the following commands.**

Now, as I am trying to replicate these solutions, like many others interested in the excellent work developed from both of you in the last years, I wonder if Jason has forgotten to update instructions or if we should still use the older version 2.88 of  bulletphysics.

It would also be very helpful to many to also know if we can work with no problems with VisualStudio say 2019 and Python say 3.8 or better stick with old ones. 

I know it is Jason's repo, but I believe you have full knowledge of this matters and could help in assisting him in fixing instructions if this is the case.

As in the next couple of years I will be deeply involved in these matters I hope in due time to be able to help writing some tutorials documenting your great efforts. For now I ask your kind suggestions.

thanks in advance 
Sergio from Milan,  another italian around beside Federico now in Manchester...

**Comment by erwincoumans** [comment]
Jason Peng's DeepMimic repo has a specific Bullet version 2.88, and he doesn't spend time updating dependencies.
https://github.com/bulletphysics/bullet3/releases/tag/2.88
I tried DeepMimic with recent Bullet versions (in double precision) and it worked just fine.

An alternative is just using PyBullet, Jason Peng used this in more recent work here (during his internship with us)
https://github.com/google-research/motion_imitation
